build.rs

```diff
@@ -134,10 +185,180 @@ fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, St
+    let subsequent_re = regex::Regex::new(r#"\s+(?:"|')([\S ]*)(?:"|')($|,|})"#)?;
+    let mut previous_finished = None;
+    for maybe_line in config_reader.lines() {
+        let line = maybe_line?;
+        if previous_finished.is_none() {
+            let captures = match entry_re.captures(&line) {
+                Some(c) => c,
+                None => continue,
+            };
+            let key = captures[1].to_owned();
+            let val = if let Some(val) = captures.get(3) {
+                val.as_str().to_owned()
+            } else {
+                captures[2].to_owned()
+            };
+            if &captures[4] != "," && &captures[4] != "}" {
+                previous_finished = Some(key.clone());
+            }
+            entries.insert(key, val);
+        } else if let Some(ref key) = previous_finished {
+            let captures = match subsequent_re.captures(&line) {
+                Some(c) => c,
+                None => continue,
+            };
+            let prev = entries.remove(key).unwrap();
+            entries.insert(key.clone(), prev + &captures[1]);
+
+            if &captures[2] == "," || &captures[2] == "}" {
+                previous_finished = None;
+            }
+        }
+    }
 
+    Ok(entries)
+}
+
+/// Find cross compilation information from sysconfigdata file
+///
+/// first find sysconfigdata file which follows the pattern [`_sysconfigdata_{abi}_{platform}_{multiarch}`][1]
+///
+/// The ABI flags can be either u, d, or m according to PEP3148. Default flags became empty since
+/// m was removed in python 3.8. (?:u|d|m|) a non capturing group for these flags
+///
+/// platform follows the output from [sys.platform][2]
+/// [a-z0-9]+
+///
+/// Multi-arch is the target triple. ([a-z_\-0-9]*)? capturing group which might not be present.
+///
+/// # Examples
+/// ```txt
+/// _sysconfigdata__freebsd_.py
+/// _sysconfigdata_m_linux_x86_64-linux-gnu.py
+/// _sysconfigdata_d_darwin_x86_64-apple-darwin.py
+/// _sysconfigdata_u_windows_i686-pc-windows-gnu.py
+/// ```
+///
+/// [1]: https://github.com/python/cpython/blob/3.8/Lib/sysconfig.py#L348
+/// [2]: https://docs.python.org/3/library/sys.html#sys.platform
+#[cfg(feature = "cross-compile")]
+fn load_cross_compile_from_sysconfigdata(
+    python_paths: PythonPaths,
+) -> Result<(InterpreterConfig, HashMap<String, String>)> {
+    let sysconfig_re =
+        regex::Regex::new(r"_sysconfigdata_(?:u|d|m|)_[a-z0-9]+_([a-z_\-0-9]*)?\.py$")?;
+    let mut walker = walkdir::WalkDir::new(&python_paths.lib_dir).into_iter();
+    let sysconfig_path = loop {
+        let entry = match walker.next() {
+            Some(Ok(entry)) => entry,
+            None => bail!("Could not find sysconfigdata file"),
+            _ => continue,
+        };
+        let entry = entry.into_path();
+        if sysconfig_re.is_match(entry.to_str().unwrap()) {
+            break entry;
+        }
+    };
```

Comment:
The point is to go recursively within the directory. Let's say the lib directory is at `/lib/`, while the `libpython3.X.so` is present at the root, the `_sysconfigdata_` file might be at `/lib/python3.x/` as it is on my manjaro system. However my cross compiled library, the file is located at `/lib/build/lib.linux-arm-3.x/` Thats why I went for a micro crate which has one dependency (or 3 if on windows) because i didn't want to do this recursive work simply for the sake of not having a dependency.

Reply:
Although if we go with the python way to get the information out of the file i'll take the `starts_with` approach so we can remove regex as a build dependency.

Reply:
please see
https://github.com/PyO3/pyo3/pull/1095/files#diff-a7b0a2dee0126cddf994326e705a91eaR260